H2MLauncher.Core/ViewModels/ServerBrowserViewModel.cs

```diff
@@ -91,8 +107,71 @@ public ServerBrowserViewModel(
             UpdateLauncherCommand = new AsyncRelayCommand(DoUpdateLauncherCommand, () => UpdateStatusText != "");
             OpenReleaseNotesCommand = new RelayCommand(DoOpenReleaseNotesCommand);
             RestartCommand = new RelayCommand(DoRestartCommand);
+            ToggleFavoriteCommand = new RelayCommand<ServerViewModel>(ToggleFavorite);
+
         }
 
+
+
+        private void AddFavoriteServer(ServerViewModel server)
+        {
+            if (server != null)
+            {
+                server.IsFavorite = true;
+                
+                if (!FavoriteServers.Select(s=>s.HostName).Contains(server.HostName))
```

Comment:
What about servers which have the same hostname? I recommend also checking the address and port